src/wp-includes/blocks.php

```diff
@@ -433,18 +435,49 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 	foreach ( $property_mappings as $key => $mapped_key ) {
 		if ( isset( $metadata[ $key ] ) ) {
 			$settings[ $mapped_key ] = $metadata[ $key ];
-			if ( $textdomain && isset( $i18n_schema->$key ) ) {
+			if ( $metadata_file_exists && $textdomain && isset( $i18n_schema->$key ) ) {
 				$settings[ $mapped_key ] = translate_settings_using_i18n_schema( $i18n_schema->$key, $settings[ $key ], $textdomain );
 			}
 		}
 	}
 
+	if ( ! empty( $metadata['render'] ) ) {
+		$template_path = $metadata_file_exists ? wp_normalize_path(
+			realpath(
+				dirname( $metadata['file'] ) . '/' .
+				remove_block_asset_path_prefix( $metadata['render'] )
+			)
+		) : '';
+		if ( $template_path ) {
+			/**
+			 * Renders the block on the server.
+			 *
+			 * @since 6.1.0
+			 *
+			 * @param array    $attributes Block attributes.
+			 * @param string   $content    Block default content.
+			 * @param WP_Block $block      Block instance.
+			 * @return string Returns the block content.
+			 */
+			$settings['render_callback'] = function( $attributes, $content, $block ) use ( $template_path ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+				ob_start();
+				require $template_path;
+				return ob_get_clean();
+			};
+		}
+	}
```

Comment:
Why is this repeated?

Reply:
It was moved in [this commit](https://github.com/WordPress/wordpress-develop/commit/35e526ed6e8e1ccbdf069ac5c30ac90db766cce6), but apparently there was a rebase conflict which caused the duplication of this code.
Fixed in https://github.com/WordPress/wordpress-develop/pull/3501/commits/280992d422e3a878aed65a9aaa36d5cf18f979d5 👍